src/main/kotlin/be/bluexin/drpc4k/jna/RPCActor.kt

```diff
@@ -0,0 +1,262 @@
+sealed class RPCInputMessage {
+    /**
+     * Make the RPC actor connect to the client.
+     *
+     * @param clientId your app's client ID.
+     * @param autoRegister whether Discord should register your app for automatic launch (untested! Probably broken because Java).
+     * @param steamId your app's Steam ID, if any.
+     * @param refreshRate the rate in milliseconds at which this handler will run callbacks and send info to discord.
+     */
+    class Connect(
+            val clientId: String,
+            val autoRegister: Boolean = false,
+            val steamId: String? = null,
+            val refreshRate: Long = 500L
+    ) : RPCInputMessage()
+
+    /**
+     * Update the user's Rich Presence.
+     * The presence will be cached if used before the app has connected, and automatically sent once ready.
+     *
+     * @param presence presence to update.
+     * @see DiscordRichPresence for all available options.
+     */
+    class UpdatePresence(val presence: DiscordRichPresence) : RPCInputMessage()
+}
+
+/**
+ * Superclass for messages sent by the RPC actor.
+ */
+@Suppress("MemberVisibilityCanBePrivate")
+sealed class RPCOutputMessage {
+    /**
+     * Sent when the RPC actor has logged in, and is ready to be accessed.
+     */
+    class Ready(val user: DiscordUser) : RPCOutputMessage()
+
+    /**
+     * Sent when the RPC actor has been disconnected.
+     *
+     * @param errorCode the error code causing disconnection.
+     * @param message the message for disconnection.
+     */
+    class Disconnected(val errorCode: Int, val message: String) : RPCOutputMessage()
+
+    /**
+     * Sent when the RPC actor has detected an error.
+     *
+     * @param errorCode the error code causing the error.
+     * @param message the message for the error.
+     */
+    class Errored(val errorCode: Int, val message: String) : RPCOutputMessage()
+
+    /**
+     * Sent when the someone accepted a game invitation.
+     *
+     * @param joinSecret the game invitation secret.
+     */
+    class JoinGame(val joinSecret: String) : RPCOutputMessage()
+
+    /**
+     * Sent when the someone accepted a game spectating invitation.
+     *
+     * @param spectateSecret the game spectating secret.
+     */
+    class Spectate(val spectateSecret: String) : RPCOutputMessage()
+
+    /**
+     * Sent when the someone requested to join the game.
+     *
+     * @param user the requester.
+     */
+    class JoinRequest(val user: DiscordUser) : RPCOutputMessage()
+}
+
+/**
+ * RPC Actor implementation.
+ *
+ * @param scope the scope for this actor to act in.
+ * @param input the actor's input channel.
+ * @param output the actor's output channel.
+ */
+@ExperimentalCoroutinesApi
+private class RPCActor(
+        private val scope: CoroutineScope,
+        private val input: ReceiveChannel<RPCInputMessage>,
+        private val output: SendChannel<RPCOutputMessage>) {
+
+    private val logger = KotlinLogging.logger { }
```

Comment:
You may move this to the top of the file so that there is only one static logger instance

Reply:
Loggers have single instance per logger name, so it won't create a new one. I feel like moving it to file might add more noise to file-level?